daemon/api_model_test.go

```diff
@@ -251,3 +291,67 @@ func (s *apiSuite) TestGetModelHasSerialAssertion(c *check.C) {
 	c.Assert(devKey, check.FitsTypeOf, "")
 	c.Assert(devKey.(string), check.Equals, string(encDevKey))
 }
+
+func (s *apiSuite) TestGetModelJSONHasSerialAssertion(c *check.C) {
+	// make a model assertion
+	theModel := s.brands.Model("my-brand", "my-old-model", modelDefaults)
+
+	deviceKey, _ := assertstest.GenerateKey(752)
+
+	encDevKey, err := asserts.EncodePublicKey(deviceKey.PublicKey())
+	c.Assert(err, check.IsNil)
+
+	// model assertion setup
+	d := s.daemonWithOverlordMock(c)
+	hookMgr, err := hookstate.Manager(d.overlord.State(), d.overlord.TaskRunner())
+	c.Assert(err, check.IsNil)
+	deviceMgr, err := devicestate.Manager(d.overlord.State(), hookMgr, d.overlord.TaskRunner(), nil)
+	c.Assert(err, check.IsNil)
+	d.overlord.AddManager(deviceMgr)
+	st := d.overlord.State()
+	st.Lock()
+	assertstatetest.AddMany(st, s.storeSigning.StoreAccountKey(""))
+	assertstatetest.AddMany(st, s.brands.AccountsAndKeys("my-brand")...)
+	s.mockModel(c, st, theModel)
+
+	// in case the name of the serial ever changes, just get it state in State
+	// currently it's hard-coded to always be serialserial
+	var authStateData auth.AuthState
+	err = st.Get("auth", &authStateData)
+	c.Assert(err, check.IsNil)
```

Comment:
this could have used d.overlord.DeviceManager().Serial() ?

Reply:
ah, I'm not sure the puzzlement from this kind of code is worth the win not to have to change the test if we change the global setup (other things might likely go awry anyway)